fvtest/compilertest/tests/MetaDataTest.cpp

```diff
@@ -0,0 +1,159 @@
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+}
+
+
+TEST(MetaDataTest, ExactInsertion) {
+   TestEnhancedCodeMetaDataManager manager;
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+
+   // Set up code cache hash table.
+   ASSERT_TRUE(NULL != manager.addFalseCacheInfo(0,1000));
+
+   TR::MethodMetaDataPOD data;
+   data.startPC = 10;
+   data.endPC   = 100;
+   data.name    = "test";
+
+   ASSERT_TRUE(manager.insertMetaData(&data));
+   ASSERT_TRUE(manager.containsMetaData(&data));
+
+   auto* metadata = manager.findMetaDataForPC(10);
+   ASSERT_TRUE(NULL != metadata);
+   ASSERT_TRUE(NULL !=  metadata->name);
+   ASSERT_STREQ("test", metadata->name);
+
+   ASSERT_TRUE(manager.removeMetaData(&data));
+}
+
+TEST(MetaDataTest, NonExactLookup) {
+   TestEnhancedCodeMetaDataManager manager;
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+
+   // Set up code cache hash table.
+   ASSERT_TRUE(NULL != manager.addFalseCacheInfo(0,1000));
+
+   TR::MethodMetaDataPOD data;
+   data.startPC = 10;
+   data.endPC   = 100;
+   data.name    = "test";
+
+   ASSERT_TRUE(manager.insertMetaData(&data));
+   ASSERT_TRUE(manager.containsMetaData(&data));
+
+   std::vector<int> test_points = {11, 25, 99};
+   for (auto i : test_points) { 
+      auto* metadata = manager.findMetaDataForPC(i);
+      ASSERT_TRUE(NULL != metadata) << "Didn't find pc" << i;
+      ASSERT_TRUE(NULL !=  metadata->name);
+      ASSERT_STREQ("test", metadata->name);
+   }
+
+   ASSERT_TRUE(NULL == manager.findMetaDataForPC(100));
+
+   ASSERT_TRUE(manager.removeMetaData(&data));
+}
+
+
+
+TEST(MetaDataTest, MultipleCaches) {
+   TestEnhancedCodeMetaDataManager manager;
+   ASSERT_TRUE(manager.initializeCodeMetaDataManager());
+
+   // Set up code cache hash table.
+   ASSERT_TRUE(NULL != manager.addFalseCacheInfo(0,1000));
+   ASSERT_TRUE(NULL != manager.addFalseCacheInfo(2000,10000));
+   ASSERT_TRUE(NULL != manager.addFalseCacheInfo(40000,90000));
+
+   TR::MethodMetaDataPOD data1;
+   data1.startPC = 10;
+   data1.endPC   = 100;
+   data1.name    = "first";
+   ASSERT_TRUE(manager.insertMetaData(&data1));
+
+   TR::MethodMetaDataPOD data2;
+   data2.startPC = 200;
+   data2.endPC   = 800;
+   data2.name    = "second";
+   ASSERT_TRUE(manager.insertMetaData(&data2));
+
+   TR::MethodMetaDataPOD data3;
+   data3.startPC = 45000;
+   data3.endPC   = 45010;
+   data3.name    = "third";
+   ASSERT_TRUE(manager.insertMetaData(&data3));
+
+   std::vector<int> test_points = {50, 300, 45005};
+   std::vector<const char*> names = {"first", "second", "third"}; 
+   int iter = 0; 
+   for (auto i : test_points) { 
+      auto* metadata = manager.findMetaDataForPC(i);
+      ASSERT_TRUE(NULL != metadata) << "Didn't find pc" << i;
+      ASSERT_TRUE(NULL !=  metadata->name);
+      ASSERT_STREQ(names[iter++], metadata->name);
+      ASSERT_TRUE(manager.removeMetaData(metadata)); 
+      ASSERT_TRUE(NULL == manager.findMetaDataForPC(i));
+   }
+
```

Comment:
Lots of "magic" numbers here (where did they come from?) so it's not really clear what the test case is trying to detect. Maybe used named values instead and add some more messages to each assert?